### PR TITLE
fix: some m3u8 downloads were broken

### LINKED ISSUE #1108: m3u8 files that do not have m3u8 in the name are ignored and will fail to download
**Metadata**
Version: [4.2.5]
OS: [PopOS]
Shell: [bash]
Anime: [Kono Subarashii Sekai ni Bakuen wo!]

**Describe the bug**
When the `download` method receives an m3u8 file without m3u8 in the name, it defaults to aria2c and fails.

**Steps To Reproduce**

1. Run `ani-cli -d -S 1 -e 1 kono-subarashii`
2. The mp4 downloaded will be an HLS playlist
3. The file aria2c attempts to download in this example is `https://news.kaavid.lol/st/ep_JhgwqSa9KnTMLhhjz_1_sub`

**Expected behavior**
There needs to be logic to check if the url is an m3u8 other than checking the extension.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Yashpal2004**
## Type of change

- [x] Bug fix

## Description

m3u8-Download-Fixed #1108

### CODE REVIEW

**Review by port19x — CHANGES_REQUESTED** [review]
bump version

File: ani-cli [hunk 1]
```diff
@@ -213,13 +213,17 @@ update_history() {
...
-    esac
+    if echo "$(curl -s "$1")" | grep -q '#EXTM3U'; then
+        ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4"
+    else
+        echo "aria $1"
+        aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide
+    fi
 }
 
+
```
Comment (port19x) [inline ani-cli:226]:
```suggestion
```

File: ani-cli [hunk 2]
```diff
@@ -213,13 +213,17 @@ update_history() {
     mv "${histfile}.new" "$histfile"
 }
 
+# Downloads a file from a URL and saves it with the specified filename in the specified directory
```
Comment (port19x) [inline ani-cli:216]:
```suggestion
```

**Comment by port19x** [comment]
Can confirm this fixes the issue

File: ani-cli [hunk 3]
```diff
@@ -213,13 +213,17 @@ update_history() {
...
-    case $1 in
-        *m3u8*) ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4" ;;
-        *) aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide ;;
-    esac
+    if echo "$(curl -s "$1")" | grep -q '#EXTM3U'; then
+        ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4"
+    else
+        echo "aria $1"
+        aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide
+    fi
```
Comment (RubenPonce) [inline ani-cli:223]: What if the url is not an m3u8 or is very large? 
Wouldn't want it to curl an entire mp4 so there should be a timeout.
```suggestion
    case $1 in
        *m3u8*) ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4" ;;
        *) if curl -s -m 5 "$1" | grep -q '^#EXTM3U'; then
        ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4"
    else
        aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide
  fi ;;
    esac
```

**Comment by Yashpal2004** [comment]
> bump version

How to do that?
I am actually new. to GIT.

File: ani-cli [hunk 4]
```diff
@@ -216,7 +216,11 @@ update_history() {
 download() {
     case $1 in
         *m3u8*) ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4" ;;
-        *) aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide ;;
+        *) if curl -s -m 5 "$1" | grep -q '^#EXTM3U'; then
+        ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4"
+    else
+        aria2c --check-certificate=false --continue --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide
+  fi ;;
```
Comment (port19x) [inline ani-cli:223]: This is pretty bad.
First you request with curl, then again with ffmpeg...

**Comment by port19x** [comment]
> 

Change the variable on the third line in ani-cli

**Review by port19x — APPROVED** [review]

**Pull request merged**